transformer_engine/pytorch/csrc/extensions/attention.cu

```diff
@@ -95,9 +95,11 @@ std::vector<at::Tensor> fused_attn_fwd_qkvpacked(
   auto qkv_sizes = QKV.sizes().vec();
   std::vector<size_t> qkv_shape{qkv_sizes.begin(), qkv_sizes.end()};
   std::vector<size_t> q_shape;
-  for (auto i : qkv_shape) {
-    if (i != 3) {
-      q_shape.push_back(i);
+  NVTE_QKV_Layout_Group layout_group = nvte_get_qkv_layout_group(qkv_layout);
+  int loc_3 = (layout_group == NVTE_3HD) ? qkv_sizes.size() - 3 : qkv_sizes.size() - 2;
```

Comment:
Should we maybe list the options explicitly here (e.g. so that we get notified with an error when we forget about changing this line)?
Something like:
```cpp
switch(layout_group) {
  case NVTE_3HD:
    loc_3 = qkv.sizes.size() - 3;
    break;
  case NVTE_H3D:  // not sure if that is the real name, did not check ;-)
    loc_3 = qkv.sizes.size() - 2;
    break;
  default:
    NVTE_ERROR("Unknown layout group");
}
```

Reply:
Done.